diff -Naur compiz-0.9.8.4-orig/kde/window-decorator-kde4/window.cpp compiz-0.9.8.4/kde/window-decorator-kde4/window.cpp
--- compiz-0.9.8.4-orig/kde/window-decorator-kde4/window.cpp	2012-09-27 12:04:02.000000000 +0100
+++ compiz-0.9.8.4/kde/window-decorator-kde4/window.cpp	2012-10-03 19:04:28.117336440 +0100
@@ -367,7 +367,7 @@
 int
 KWD::Window::tabCount () const
 {
-    return 0;
+    return 1;
 }
 
 long
